includes/class-wcsatt-scheme.php

```diff
@@ -0,0 +1,431 @@
+		'subscription_pricing_method'  => 'pricing_mode',
+		'subscription_discount'        => 'discount',
+		'subscription_regular_price'   => 'regular_price',
+		'subscription_sale_price'      => 'sale_price',
+		'subscription_price'           => 'price'
+	);
+
+	/**
+	 * Constructor. Currently only initializes the object from raw data.
+	 * Later, it could initialize using other source data, such as a DB ID.
+	 *
+	 * @param  array  $args
+	 */
+	public function __construct( $args ) {
+
+		if ( isset( $args[ 'data' ] ) ) {
+
+			$this->data[ 'period' ]       = isset( $args[ 'data' ][ 'subscription_period' ] ) ? strval( $args[ 'data' ][ 'subscription_period' ] ) : '';
+			$this->data[ 'interval' ]     = isset( $args[ 'data' ][ 'subscription_period_interval' ] ) ? absint( $args[ 'data' ][ 'subscription_period_interval' ] ) : '';
+			$this->data[ 'length' ]       = isset( $args[ 'data' ][ 'subscription_length' ] ) ? absint( $args[ 'data' ][ 'subscription_length' ] ) : '';
+
+			$this->data[ 'trial_period' ] = isset( $args[ 'data' ][ 'subscription_trial_period' ] ) ? strval( $args[ 'data' ][ 'subscription_trial_period' ] ) : '';
+			$this->data[ 'trial_length' ] = isset( $args[ 'data' ][ 'subscription_trial_length' ] ) ? absint( $args[ 'data' ][ 'subscription_trial_length' ] ) : '';
+
+			$this->data[ 'pricing_mode' ] = isset( $args[ 'data' ][ 'subscription_pricing_method' ] ) && in_array( $args[ 'data' ][ 'subscription_pricing_method' ], array( 'inherit', 'override' ) ) ? strval( $args[ 'data' ][ 'subscription_pricing_method' ] ) : 'inherit';
+
+			if ( 'override' === $this->data[ 'pricing_mode' ] ) {
+				$this->data[ 'regular_price' ] = isset( $args[ 'data' ][ 'subscription_regular_price' ] ) ? wc_format_decimal( $args[ 'data' ][ 'subscription_regular_price' ] ) : '';
+				$this->data[ 'sale_price' ]    = isset( $args[ 'data' ][ 'subscription_sale_price' ] ) ? wc_format_decimal( $args[ 'data' ][ 'subscription_sale_price' ] ) : '';
+				$this->data[ 'price' ]         = '' !== $this->data[ 'sale_price' ] && $this->data[ 'sale_price' ] < $this->data[ 'regular_price' ] ? $this->data[ 'sale_price' ] : $this->data[ 'regular_price' ];
+
+				if ( '' === $this->data[ 'price' ] && '' === $this->data[ 'regular_price' ] ) {
+					$this->data[ 'pricing_mode' ] = 'inherit';
+				}
+			}
+
+			if ( 'inherit' === $this->data[ 'pricing_mode' ] ) {
+				$this->data[ 'discount' ] = isset( $args[ 'data' ][ 'subscription_discount' ] ) ? wc_format_decimal( $args[ 'data' ][ 'subscription_discount' ] ) : '';
+			}
+		}
+
+		$this->data[ 'context' ] = isset( $args[ 'context' ] ) ? strval( $args[ 'context' ] ) : 'product';
+
+		$this->key = implode( '_', array_filter( array( $this->data[ 'interval' ], $this->data[ 'period' ], $this->data[ 'length' ] ) ) );
+	}
+
+	/**
+	 *
+	 * Returns a unique scheme identifier. For now it just returns the key. Later it could be a CRUD object identifier.
+	 *
+	 * @return string  A unique identifier.
+	 */
+	public function get_id() {
```

Comment:
I don't think `get_id()` is used anywhere. Agreed it's better to remove it. Added to #151

> Then if it moves to extend `WC_Data`, we could deprecate `get_key()` and provide `get_id()` without changing its return value.

In this scenario, CRUD IDs would be most likely BIGINT, while keys would still be used as string representations of schemes. Doesn't matter at this time but I have the impression we'd need to keep `get_key`.